app/abci.go

```diff
@@ -812,3 +826,25 @@ func (app *HeimdallApp) PreBlocker(ctx sdk.Context, req *abci.RequestFinalizeBlo
 
 	return app.ModuleManager.PreBlock(ctx)
 }
+
+func (app *HeimdallApp) getValidatorSetForHeight(ctx sdk.Context, height int64) (*stakeTypes.ValidatorSet, error) {
+	var (
+		validatorSet *stakeTypes.ValidatorSet
+		err          error
+	)
+
+	if height >= helper.GetTallyFixHeight() && height >= helper.GetInitialHeight()+2 {
```

Comment:
Why do we need `height >= helper.GetInitialHeight()+2` in this IF?

IIUC, `TallyFixHeight` is way greater than `InitialHeight`... so the second condition is trivially true if the first one is true

Reply:
Yes this is true for production but for unit tests and devnets we need to have this condition.